lib/src/app/cast.js

```diff
@@ -17,59 +19,73 @@ import '../styles/player.css';
 
-    // create a new instance of the Media Manager to interact with the sender app request
-    mediaManager = new cast.receiver.MediaManager(castPlayer);
+  const castPlayer = new CastPlayer(namespace, pageLevelParameters);
+  const mediaManager = new cast.receiver.MediaManager(castPlayer);
 
-    // backup of the original onLoad event handler to use it after our custom logic
+  /**
+   * Exotic class method extending by Google devs
+   * @see https://developers.google.com/cast/docs/custom_receiver (section Media)
+   * onLoad assigns a new implementation of the onLoad event that can set the
+   * needed information from the incoming event
+   * @param {cast.receiver.MediaManager.Event} event
+   */
+  mediaManager.onLoad = (() => {
     mediaManager.origOnLoad = mediaManager.onLoad;
-
-    /**
-     * onLoad assign a new implementation of the onLoad event that can sets the needed information from the
-     * incoming event
-     * @param {cast.receiver.MediaManager.Event} event
-     */
-    mediaManager.onLoad = function(event) {
-        logger.info('MediaManager:onLoad', event);
-        if (event && event.data && event.data.autoplay) {
-            Timer.setIdle();
-        }
-        castPlayer.setAsset(event.data.media.customData);
-        if (castPlayer.OOPlayer != null) {
-            castPlayer.notifySenders(castPlayer.OOPlayer.getState());
+    return (event) => {
+      logger.info('MediaManager:onLoad', event);
+      const isAutoplayDesired = event && event.data && event.data.autoplay;
+      if (isAutoplayDesired) {
+        Timer.setIdle();
+      }
+      castPlayer.setAsset(event.data.media.customData);
+      if (castPlayer.OOPlayer != null) {
+        const state = castPlayer.OOPlayer.getState();
+        castPlayer.notifySenders(state);
+        if (isAutoplayDesired) {
+          castPlayer.play();
         }
-        mediaManager.origOnLoad(event);
+      }
+      mediaManager.origOnLoad(event);
     };
+  })();
 
-    // backup of the original onPause event handler to use it after our custom logic
-    mediaManager.origPause = mediaManager.onPause;
-
-    /**
-     * onPause overrides the default behaviour to setup the default timeout when an assets it's getting paused
-     */
-    mediaManager.onPause = function(e) {
-        logger.info('MediaManager:onPause', e);
-        Timer.setIdle(Timer.TIMEOUT.PAUSED);
-        mediaManager.origPause(e);
+  /**
+   * Exotic class method extending by Google devs
+   * @see https://developers.google.com/cast/docs/custom_receiver (section Media)
+   * onPause overrides the default behaviour to setup the default timeout
+   * when an asset is being paused
+   * @param {cast.receiver.MediaManager.Event} event
+   */
+  mediaManager.onPause = (() => {
+    mediaManager.origOnPause = mediaManager.onPause;
+    return (event) => {
+      logger.info('MediaManager:onPause', event);
+      Timer.setIdle(Timer.TIMEOUT.PAUSED);
+      if (castPlayer.OOPlayer != null) {
```

Comment:
This also looks to be a logic change.

Reply:
This is just one more way to extend an object created in ugly way be a google team: https://developers.google.com/cast/docs/custom_receiver (method of the object had to be implemented via prototype; it this case extend would go soft; but google team done awkward object and suggests now a tricky way to extend it, oops).
Logic doesn't change here

Reply:
Sounds good, thanks for the explanation!